ecosystem/gstreamer_plugin/gst_mtl_st30p_tx.c

```diff
@@ -199,6 +213,9 @@ static void gst_mtl_st30p_tx_init(Gst_Mtl_St30p_Tx* sink) {
   gst_pad_set_event_function(sinkpad, GST_DEBUG_FUNCPTR(gst_mtl_st30p_tx_sink_event));
 
   gst_pad_set_chain_function(sinkpad, GST_DEBUG_FUNCPTR(gst_mtl_st30p_tx_chain));
+
+  pthread_mutex_init(&sink->session_mutex, NULL);
```

Comment:
ok this is a question 

why are we doing this "always" 

Shoudn't this mutex by only initialized when with the parameter ? 

we also set the status to ready always ?
Im not opposed to this I'm curious for reasons 

cuz it's only "used" with arguments shoudln't all interactions with this be conditional ?

Reply:
this doesn't make any difference, but ok, in future if we want to remove the feature it will be easier.

Reply:
true -> i don't insist on it 

i see both sides of this argument 
-> increases simplicity when not implemented all the if makes the code cluttered 

but on the other hand when you see that there is a ready statement in the code with the if statement the functionality is immidielty apparent, you understand where it's comes from

do you agree ? 
I don't think it's madness to do that